integration/lifecycle/capabilities_endorse_test.go

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package lifecycle
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger/fabric/integration/channelparticipation"
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/raft"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/ginkgomon_v2"
+)
+
+var _ = Describe("Lifecycle with Channel v3_0 capabilities and ed25519 identities", func() {
+	var (
+		client       *docker.Client
+		testDir      string
+		network      *nwo.Network
+		ordererProcs []ifrit.Process
+		peerProcs    []ifrit.Process
+		channelID    string
+		chaincode    nwo.Chaincode
+	)
+
+	BeforeEach(func() {
+		var err error
+		testDir, err = os.MkdirTemp("", "lifecycle")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+		channelID = "testchannel"
+
+		chaincode = nwo.Chaincode{
+			Name:            "mycc",
+			Version:         "0.0",
+			Path:            "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Lang:            "golang",
+			PackageFile:     filepath.Join(testDir, "simplecc.tar.gz"),
+			Ctor:            `{"Args":["init","a","100","b","200"]}`,
+			SignaturePolicy: `AND ('Org1MSP.peer', 'Org2MSP.peer')`,
+			Sequence:        "1",
+			InitRequired:    true,
+			Label:           "my_simple_chaincode",
+		}
+	})
+
+	AfterEach(func() {
+		// Shutdown processes and cleanup
+		allProcs := append(ordererProcs, peerProcs...)
+		for _, process := range allProcs {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	It("invoke chaincode after upgrading Channel to V3_0 and add ed25519 peer and orderer", func() {
+		networkConfig := nwo.MultiNodeEtcdRaft()
```

Comment:
So we have a single organization with 2 orderers of ECDSA and 1 orderer that is offline, and later is brought online to be ed25519 ?

Reply:
Almost that. Initially, we have 3 ECDSA orderers from a single organization. Later, orderer3 is killed, given an ed25519 key, and restarted. In contrast, there are 2 ECDSA peers from different orgs and an ed25519 offline one. The ed25519 peer is only started once, directly with ed25519 keys.